src/libraries/System.Net.Http/src/System/Net/Http/HttpClient.cs

```diff
@@ -508,9 +508,16 @@ private void CheckRequestBeforeSend(HttpRequestMessage request)
             try
             {
                 // Wait for the send request to complete, getting back the response.
-                response = async ?
-                    await base.SendAsync(request, cts.Token).ConfigureAwait(false) :
-                    base.Send(request, cts.Token);
+                if (async)
+                {
+                    response = await base.SendAsync(request, cts.Token).ConfigureAwait(false);
+                }
+                else
+                {
+                    Debug.Assert(!OperatingSystem.IsBrowser());
```

Comment:
This assert seems problematic.  A caller of HttpClient.Send might get a warning but could still choose to make the call, at which point the assert would fire: I'd expect we'd want tests to validate that HttpClient.Send threw an exception for browser, and it's going to hit this assert in debug builds.

Reply:
Oh, interesting; my bad on that. Then perhaps we should just revert back to how the code was and keep the `#pragma`, including a comment that the code is reachable, but is expected to throw at runtime.

Reply:
> A caller of HttpClient.Send might get a warning but could still choose to make the call, at which point the assert would fire: I'd expect we'd want tests to validate that HttpClient.Send threw an exception for browser, and it's going to hit this assert in debug builds, crashing the test process

Good point, I don't know if there is a test for the browser (in runtime), most likely not. But I expect if a user's project is targeting a browser and they had a test then they would see the failure. 

> Asserts should only be used for things that should never be possible.

yes and plus, it is also could be used to make sure the developer assumption